File: clean_html.py
import re

with open('src/main/resources/templates/battle-arena.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Remove all HTML comments that contain strange repeated characters
content = re.sub(r'<!--[ \n]*A[,A\?\+T\.\_a-zA-Z\s]+-->', '', content)
content = re.sub(r'<!--\s*(A.*?|A\?\?.*?|A,A.*?)\s*-->', '', content, flags=re.DOTALL)

# Also remove inline garbage text if it's sitting as a direct text node
# Looking for 'A' followed by  or ? or , repeated
content = re.sub(r'A[\?\,\+T]{2,}\w*', '', content)

with open('src/main/resources/templates/battle-arena.html', 'w', encoding='utf-8') as f:
    f.write(content)

File: clean_html.js
const fs = require('fs');
let content = fs.readFileSync('src/main/resources/templates/battle-arena.html', 'utf-8');

// Remove all HTML comments completely EXCEPT those containing standard text
content = content.replace(/<!--[\s\S]*?-->/g, match => {
    // If the comment has a lot of non-ascii or repeated A patterns, delete it
    if (match.includes('A') || match.includes('A,A') || match.includes('A?A') || match.match(/A[\?,+T]{2,}/)) {
        return '';
    }
    return match;
});

// Also fix inline garbage in Hero Banner
content = content.replace(/<h1>.*?Battle Arena<\/h1>/, '<h1>Battle Arena</h1>');
content = content.replace(/Your vote has been recorded!.*?<\/div>/s, 'Your vote has been recorded!</div>');

fs.writeFileSync('src/main/resources/templates/battle-arena.html', content, 'utf-8');
